### PR TITLE
Group Metadata

### PR DESCRIPTION
**Pull request by Guillar1**

### CODE REVIEW

File: src/grouping-and-mapping/clients/GroupsClient.ts [hunk 1]
```diff
@@ -73,38 +78,52 @@ export class GroupsClient extends OperationsBase  implements IGroupsClient {
...
+    const request = this.createRequest("GET", accessToken, undefined, preferReturn);
+
+    if (preferReturn === PreferReturn.Representation) {
+      return new EntityListIteratorImpl(async () => this.fetchCollection<Group>(url, request));
+    } else {
+      return new EntityListIteratorImpl(async () => this.fetchCollection<GroupMinimal>(url, request));
+    }
+  }
+
+  protected async fetchCollection<T extends GroupMinimal | Group>(url: string, request: RequestInit): Promise<EntityCollectionPage<T>> {
```
Comment (MindaugasButkus) [inline src/grouping-and-mapping/clients/GroupsClient.ts:118]: Why protected?

File: src/common/OperationsBase.ts [hunk 2]
```diff
@@ -118,4 +124,21 @@ export class OperationsBase {
   protected topIsValid(top: number | undefined): boolean {
     return top !== undefined ? (top > 0 && top <= 1000) : true;
   }
+
+  /**
+   * Validates metadata entries.
+   * @param entries
+   */
+  protected validateMetadata(entries: GroupMetadata[]): void {
```
Comment (MindaugasButkus) [inline src/common/OperationsBase.ts:132]: No reason for this to be in the OperationsBase. Move to GroupsClient.

File: src/grouping-and-mapping/clients/GroupsClient.ts (hunk 1 above)
Comment (Guillar1) [reply src/grouping-and-mapping/clients/GroupsClient.ts]: Maybe for future extensibility (not likely). But private makes more sense now.

File: src/common/OperationsBase.ts (hunk 2 above)
Comment (Guillar1) [reply src/common/OperationsBase.ts]: Moved.

**Review by MindaugasButkus — APPROVED** [review]

**Pull request merged**